src/main/java/seedu/workbook/logic/parser/FindCommandParser.java

```diff
@@ -16,18 +23,65 @@ public class FindCommandParser implements Parser<FindCommand> {
     /**
      * Parses the given {@code String} of arguments in the context of the FindCommand
      * and returns a FindCommand object for execution.
+     *
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
+        requireNonNull(args);
+
+
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_COMPANY, PREFIX_ROLE,
+                PREFIX_STAGE);
+
+        int numberOfPrefixes = ArgumentTokenizer.numberOfPrefixes(args, PREFIX_COMPANY, PREFIX_ROLE,
+                PREFIX_STAGE);
+
+        if (!argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+
+        } else if (numberOfPrefixes > 1) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
-        }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        } else if (argMultimap.getValue(PREFIX_COMPANY).isPresent()) {
+            String arguments = argMultimap.getValue(PREFIX_COMPANY).get();
+            if (arguments.isEmpty()) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+            }
+            String[] nameKeywords = arguments.split(" ");
```

Comment:
any reason why you changed the splitting from `.split("\\s+")`?
from what I read, that regex means to split by one or more white space